observability/metrics/src/main/scala/com/daml/metrics/InstrumentedExecutorServiceMetrics.scala

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.metrics
+
+import java.util
+import java.util.concurrent.{Callable, ExecutorService, Future, TimeUnit}
+
+import com.daml.metrics.api.MetricDoc.MetricQualification.Debug
+import com.daml.metrics.api.MetricHandle.{Counter, Meter, MetricsFactory, Timer}
+import com.daml.metrics.api.{MetricDoc, MetricsContext}
+
+import scala.jdk.CollectionConverters.{CollectionHasAsScala, SeqHasAsJava}
+
+class InstrumentedExecutorServiceMetrics(factory: MetricsFactory) {
+
+  @MetricDoc.Tag(
+    summary = "The number of tasks submitted to an instrumented executor.",
+    description = "Number of tasks that were submitted to the executor.",
+    qualification = Debug,
+    labelsWithDescription = Map(
+      "name" -> "The name of the executor service.",
+      "type" -> "The type of the executor service. Can be `fork_join` or `thread_pool`.",
+    ),
+  )
+  val submitted: Meter =
+    factory.meter(
+      InstrumentedExecutorServiceMetrics.Prefix :+ "submitted",
+      "Number of tasks submitted to this executor",
+    )
+
+  @MetricDoc.Tag(
+    summary = "The number of tasks running in an instrumented executor.",
+    description = "Currently running number of tasks.",
```

Comment:
Just wondering: do we want to make an attempt at aligning the metric descriptions reported to Prometheus and the description/summary used for metric documentation?
These are more or less aligned currently (i.e., even if they don't have the exact wording), however, we might want to avoid divergence to not confuse consumers of these different pieces of information (e.g., when someone looks at the description on Grafana vs. at the public documentation).

Reply:
Sure, let me align the new metrics at least for now.

Reply:
Sync'ed